services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -191,6 +251,74 @@ class HankeServiceImpl (@Autowired val hankeRepository: HankeRepository) : Hanke
             hanke.meluHaitta?.let { entity.meluHaitta = hanke.meluHaitta }
             hanke.polyHaitta?.let { entity.polyHaitta = hanke.polyHaitta }
             hanke.tarinaHaitta?.let { entity.tarinaHaitta = hanke.tarinaHaitta }
+
+        }
+
+        ///method combines three lists to one for database
+        private fun createEntityFromHankeYhteystiedotDomainObject(h: Hanke, hankeEntity: HankeEntity) {
+
+            hankeEntity.listOfHankeYhteystieto = mutableListOf<HankeYhteystietoEntity>()
+
+            addHankeYhteystietoEntitysToList(h.omistajat, hankeEntity)
+            addHankeYhteystietoEntitysToList(h.arvioijat, hankeEntity)
+            addHankeYhteystietoEntitysToList(h.toteuttajat, hankeEntity)
         }
+
+        private fun addHankeYhteystietoEntitysToList(listOfHankeYhteystiedot: List<HankeYhteystieto>, hankeEntity: HankeEntity) {
+            val iterator = listOfHankeYhteystiedot.iterator()
+
+            while (iterator.hasNext()) {
+
+                var hankeYht: HankeYhteystieto = iterator.next()
+
+                var hankeYhtEnt: HankeYhteystietoEntity = HankeYhteystietoEntity(
+                        hankeYht.contactType,
+                        hankeYht.sukunimi,
+                        hankeYht.etunimi,
+                        hankeYht.email,
+                        hankeYht.puhelinnumero,
+                        hankeYht.organisaatioId,
+                        hankeYht.organisaatioNimi,
+                        hankeYht.osasto,
+                        1, //TODO: real user
+                        hankeYht.createdAt?.toLocalDateTime(),
+                        null, //TODO: real user
+                        getCurrentTimeUTCAsLocalTime(),  //TODO: if changed?
```

Comment:
I left these created and modified -things out of the entity-creation, specifically so that the calling method (create vs. update) can choose what to do. However, for the main Hanke, there is only one entity, so it is easy. Here one request could end up having 10+ Yhteystieto-entities, so it might be easier to just give the desired created/modified values as parameters to the function call. Especially since we do not want to resolve the user-value 10+ times during the same request.

Reply:
This is a bit tricky in many ways. Do we want to update the modifiedat and by also when they maybe did not change? We need to come back to this when we have the User and we are actually implementing update functionality by another user.

I'll add a TODO-comment for it